cuckoo/finalizers.py

```diff
@@ -183,38 +183,44 @@ class YieldingFinalizer(
     Generic[TYIELD],
 ):
     def __init__(
-        self: YieldingFinalizer,
+        self,
         node: BinaryTreeNode,
-        returning_fn: Callable[[], Generator[TYIELD, None, None]],
         gen_to_val: dict,
+        returning_fn: Optional[Callable[[], Generator[TYIELD, None, None]]] = None,
+        streaming_fn=None,
         response_key=ColumnResponseKey,
         **kwargs,
     ):
         super().__init__(node=node, **kwargs)
+        self._gen_to_val = gen_to_val
         self._returning_fn = returning_fn
+        self._streaming_fn = streaming_fn
         self._response_key = response_key
-        self._gen_to_val = gen_to_val
 
     def yielding(
-        self: YieldingFinalizer,
+        self,
         columns: Optional[TCOLUMNS] = None,
         *,
         invert_selection=False,
     ) -> Generator[TYIELD, None, None]:
         resolved_columns = self._resolve_column_selection(columns, invert_selection)
         self._node._fragments.response_keys.append(self._response_key(resolved_columns))
         ExecutingFinalizer._bind_includes(self._node)
-        self._execute()
 
-        return self._returning_fn()
+        if self._streaming_fn:
+            self._execute(stream=True)
+            return self._streaming_fn()
+        else:
+            self._execute()
+            return self._returning_fn()
```

Comment:
part 1 of the feature: if the query builder has a streaming handler, we want to use it. for all other builders, i.e. `Query().one_by_pk().yielding()`, `Update().many().yielding()`, `Delete().many().yielding()` etc, we use the old code unchanged